File: Strings/Q4.cpp
#include<iostream>
using namespace std;

void diffOfTwo(string str)
{
    string ans;
    ans += str[0];
    for(int i = 1; i < str.length(); i++)
    {
        int diff = str[i] - str[i-1];
        char ch = str[i];
        ans += to_string(diff);
        ans += ch;
    }
    cout << ans;
}

int main()
{
    string str;
    cin >> str;
    diffOfTwo(str);
}

File: Strings/Q4.java
import java.util.Scanner;
public class Q4
{
    public static Scanner scn = new Scanner(System.in);
    public static void diffOfTwo(String str)
    {
        StringBuilder sb = new StringBuilder();
        sb.append(str.charAt(0));
        for(int i = 1; i < str.length(); i++)
        {
            int diff = str.charAt(i) - str.charAt(i-1);
            char ch = str.charAt(i);
            sb.append(diff);
            sb.append(ch);
        }
        System.out.println(sb.toString());
    }

    public static void main(String[] args)
    {
        String str = scn.nextLine();
        diffOfTwo(str);
    }
}